tools/copy_bom/src/util.rs

```diff
@@ -263,11 +310,30 @@ pub fn extract_dependencies_from_output(
         if let Some(captures) = captures {
             let raw_path = (&captures["path"]).to_string();
             if let Some(absolute_path) = convert_to_absolute(file_path, &raw_path) {
-                shared_objects.insert((absolute_path.clone(), absolute_path.clone()));
-                let raw_name = (&captures["name"]).to_string();
-                let raw_name_path = PathBuf::from(&raw_name);
-                if raw_name_path.is_absolute() {
-                    shared_objects.insert((absolute_path, raw_name));
+                match default_lib_dirs {
+                    None => {
+                        shared_objects.insert((absolute_path.clone(), absolute_path));
+                    }
+                    Some(ref default_lib_dirs) => {
+                        let file_name = (&captures["name"]).to_string();
+                        let lib_dir_in_host = PathBuf::from(&absolute_path)
+                            .parent()
+                            .unwrap()
+                            .to_string_lossy()
+                            .to_string();
+                        // if the shared object is from one of the default dirs, we will copy to the first default dir
+                        // Otherwise it will be copied to the same dir as its dir in host.
+                        if default_lib_dirs.contains(&lib_dir_in_host) {
```

Comment:
I think if the first default directory is so special, we should maintain a row for this value.

Reply:
fixed. Now, all libraries in these paths will be copied to the loader directory.